maestro-ios/src/main/java/ios/xctest/XCTestIOSDevice.kt

```diff
@@ -169,7 +175,8 @@ class XCTestIOSDevice(
     }
 
     override fun stop(id: String): Result<Unit, Throwable> {
-        error("Not supported")
+        deviceId?.let { deviceId -> Simctl.terminate(deviceId, id) }
```

Comment:
I don't think this fits in `XCTestIOSDevice` - there is a dependency on Simctl that we can't satisfy in Cloud. I suggest creating a new `SimctlIOSDevice` that has those methods implemented in there and route to them from `LocalIOSDevice`

Reply:
Having many (3) half-implemented classes does not help with programming safety.
I thought of XCTestIOSDevice as the 'idb-less future', so the unimplemented methods make sense because we're in a transition period.
I'm opposed to make a permanent setup where half of the methods are not implemented and the caller needs to understand what implementation it is talking to to safely call a method.

Reply:
One way forward is to drop the XCTestIOSDevice: IOSDevice adherence (rename to XCTestIOSHelper?), and create a SimcltIOSHelper which implements the simctl code and also does not adhere to IOSDevice.

Reply:
That approach did not work in the past as we ended up with a bunch of mix-ins in an IOSDevice implementation, some of which worked in Cloud and some didn’t.

Half-implemented versions of this class allow us to clearly separate at Local/Cloud device level what implementation is expected to work when. I.e. in current setup calling XCTest device is expected to work in cloud for all implemented method except the ones that were just added. Let’s keep it consistent - any given implementation should either fully support cloud or not support it at all.

Reply:
I'll go forward with the SimctlIOSDevice approach you suggested for now and see if I can find an architectural improvement separately.